cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1629,13 +1630,67 @@ private String generatePerBrokerConfiguration(NodeRef node, KafkaPool pool, Map<
                         .withAuthorization(cluster, authorization)
                         .withCruiseControl(cluster, ccMetricsReporter, node.broker())
                         .withUserConfiguration(configuration, node.broker() && ccMetricsReporter != null);
+        buildBrokerConfiguration(node, pool, builder);
+        return builder.build().trim();
+    }
 
-        if (useKRaft) {
-            builder.withKRaft(cluster, namespace, pool.processRoles, nodes());
-        } else {
-            builder.withZookeeper(cluster);
+    /**
+     * Adds ZooKeeper and/or KRaft connection and/or ZooKeeper migration depending on the role of the node (broker or controller)
+     * and the Kafka metadata configuration state.
+     * This method actually implements the following table.
+     *
+     * +----------------+--------------+--------------+
+     * |                | Controller   | Broker       |
+     * +----------------+--------------+--------------+
+     * | ZK             | -            | Zk           |
+     * +----------------+--------------+--------------+
+     * | PRE_MIGRATION  | KRaft        | Zk           |
+     * |                | Zk           |              |
+     * |                | Zk-migration |              |
+     * +----------------+--------------+--------------+
+     * | MIGRATION      | KRaft        | KRaft        |
+     * |                | Zk           | Zk           |
+     * |                | Zk-migration | Zk-migration |
+     * +----------------+--------------+--------------+
+     * | POST_MIGRATION | KRaft        | KRaft        |
+     * |                | Zk           |              |
+     * |                | Zk-migration |              |
+     * +----------------+--------------+--------------+
+     * | KRAFT          | KRaft        | KRaft        |
+     * +----------------+--------------+--------------+
+     *
+     * @param node node on which the configuration is applied
+     * @param pool pool hosting the node
+     * @param builder KafkaBrokerConfigurationBuilder instance to use to build the node configuration
+     */
+    private void buildBrokerConfiguration(NodeRef node, KafkaPool pool, KafkaBrokerConfigurationBuilder builder) {
```

Comment:
This method is super hard to read. You should at least add some spacing. The INFO messages seem also a bit too much?

Reply:
Not sure where you think spacing makes it more readable, if it's really hard to read.
Regarding the INFO logging, it will become TRACE level in the end. Currently, I am using INFO for having a better testing feedback without the need to increase the overall debugging level of the operator.

Reply:
The block has 50 lines of various ifs without a single empty line between them. That makes it hard to read.